app/src/main/java/com/example/vybe/ViewVibeActivity.java

```diff
@@ -61,38 +62,48 @@ protected void onCreate(Bundle savedInstanceState) {
 
         if (extras.containsKey("vibeEvent")) {
             vibeEvent = (VibeEvent) extras.getSerializable("vibeEvent");
+            populateVibeEventDetails(vibeEvent);
+        }
 
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(" d, YYYY h:mm a", Locale.ENGLISH);
-            LocalDateTime dateTime = vibeEvent.getDateTime();
-            String month = dateTime.getMonth().getDisplayName(TextStyle.FULL_STANDALONE, Locale.ENGLISH);
-            dateField.setText(month + dateTime.format(formatter));
-            String reason = vibeEvent.getReason();
-            String socialSituation = vibeEvent.getSocialSituation();
-            // TODO: missing location - do that later once done
-            String reasonImagePath = vibeEvent.getImage();
-
-            if (reasonImagePath != null){
-                loadImageFirebase(reasonImage, reasonImagePath);
-            }
+    }
 
-            if (vibeEvent.getVibe() != null) {
-                vibeImage.setImageResource(vibeEvent.getVibe().getEmoticon());
-                toolbar.setBackgroundResource(vibeEvent.getVibe().getColor());
-            }
 
-            if (reason == null || reason.equals("")) {  // Reason is optional
-                reasonLabel.setVisibility(TextView.GONE);
-                reasonField.setVisibility(TextView.GONE);
-            } else {
-                reasonField.setText(reason);
-            }
+    /**
+     * This will populate the appropriate fields for viewing Vibe Event details
+     * @param vibeEvent
+     *      The Vibe Event object containing details to be displayed
+     */
+    public void populateVibeEventDetails(VibeEvent vibeEvent) {
+        //
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(" d, YYYY h:mm a", Locale.ENGLISH);
+        LocalDateTime dateTime = vibeEvent.getDateTime();
+        String month = dateTime.getMonth().getDisplayName(TextStyle.FULL_STANDALONE, Locale.ENGLISH);
+        String dateFieldText = month + dateTime.format(formatter);
+        dateField.setText(dateFieldText);
+        String reason = vibeEvent.getReason();
+        String socialSituation = vibeEvent.getSocialSituation();
+        // TODO: missing location - do that later once done
+        String reasonImagePath = vibeEvent.getImage();
+
+        if (reasonImagePath != null){
+            loadImageFirebase(reasonImage, reasonImagePath);
+        }
```

Comment:
I wrote this function but we should keep in mind populating vibe event details is common and done through a bunch of different code, so we could move this to a controller I guess

Reply:
Kk